arch/riscv/core/irq_manage.c

```diff
@@ -51,4 +52,11 @@ int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
 #endif
 	return irq;
 }
+
+int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
+			     void (*routine)(const void *parameter),
+			     const void *parameter, uint32_t flags)
+{
+	return z_soc_irq_connect_dynamic(irq, priority, routine, parameter, flags);
```

Comment:
Other riscv "z_soc_irq_xxx" stuff are behind a CONFIG_RISCV_SOC_HAS_CUSTOM_XXX - why not follow the same pattern here?